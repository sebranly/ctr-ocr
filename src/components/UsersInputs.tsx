import * as React from 'react';

export interface UsersInputsProps {
  isDisabledUI: boolean;
  suggestions: string[];
  players: string[];
  setPlayers: React.Dispatch<React.SetStateAction<string[]>>;
}

const UsersInputs: React.FC<UsersInputsProps> = (props) => {
  const { isDisabledUI, players, setPlayers, suggestions } = props;

  const onChangeInput = (index: number) => (e: React.ChangeEvent<HTMLInputElement>) => {
    if (players.length <= index) return;

    const { value } = e.currentTarget;
    const newPlayers = [...players];

    newPlayers[index] = value;

    setPlayers(newPlayers);
  };

  const renderUserInput = (index: number) => {
    const datalistIdString = `human-player-suggestion-${index}`;
    return (
      <div key={index}>
        <input
          className="mb mt big-input"
          disabled={isDisabledUI}
          list={datalistIdString}
          onChange={onChangeInput(index)}
          value={players[index] || ''}
        />
        {renderDatalist(index)}
      </div>
    );
  };

  const renderDatalist = (datalistId: number) => {
    const datalistIdString = `human-player-suggestion-${datalistId}`;
    return (
      <datalist id={datalistIdString}>
        {suggestions.map((suggestion) => (
          <option key={suggestion} value={suggestion} />
        ))}
      </datalist>
    );
  };

  return <div>{players.map((_players, index) => renderUserInput(index))}</div>;
};

export { UsersInputs };
