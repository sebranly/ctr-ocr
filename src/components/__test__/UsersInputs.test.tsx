import * as React from 'react';
import { render } from '@testing-library/react';

import { UsersInputs } from '../UsersInputs';

test('UsersInputs', () => {
  const createComponent = () => (
    <UsersInputs
      isDisabledUI={false}
      players={['player1', 'player2', 'player3']}
      setPlayers={() => {}}
      suggestions={['someone', 'else']}
    />
  );

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
