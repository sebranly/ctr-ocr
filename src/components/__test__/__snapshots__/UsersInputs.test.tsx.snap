// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersInputs 1`] = `
NodeList [
  <div>
    <div>
      <input
        class="mb mt big-input"
        list="human-player-suggestion-0"
        value="player1"
      />
      <datalist
        id="human-player-suggestion-0"
      >
        <option
          value="someone"
        />
        <option
          value="else"
        />
      </datalist>
    </div>
    <div>
      <input
        class="mb mt big-input"
        list="human-player-suggestion-1"
        value="player2"
      />
      <datalist
        id="human-player-suggestion-1"
      >
        <option
          value="someone"
        />
        <option
          value="else"
        />
      </datalist>
    </div>
    <div>
      <input
        class="mb mt big-input"
        list="human-player-suggestion-2"
        value="player3"
      />
      <datalist
        id="human-player-suggestion-2"
      >
        <option
          value="someone"
        />
        <option
          value="else"
        />
      </datalist>
    </div>
  </div>,
]
`;
